### PR TITLE
[compiler-v2] Change compiler abort codes to conform with std::errors

### PR DESCRIPTION
**Pull request by wrwg**
## Description

The module `std::error` in the standard library defines a mechanism to organize errors, which we also use in the framework. This adapts the compiler generated abort codee (for assert and incomplete match) to this discipline. The docs have been already updated in a pending PR.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?

Existing tests

### CODE REVIEW

**Comment by wrwg** [comment]
* **#14457** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/14457?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @wrwg and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by rahxephon89 — APPROVED** [review]

File: third_party/move/move-model/src/well_known.rs [hunk 1]
```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
...
 pub const RECEIVER_PARAM_NAME: &str = "self";
 
-/// The well-known abort codes used by the compiler. These are marked
-/// by lowest 6 bytes of a sha256 of the string "Move 2 Abort Code",
-/// appended with two bytes for the error type.
-/// TODO: add a check at runtime that user is not clashing with reserved
-/// codes?
-pub const WELL_KNOWN_ABORT_CODE_BASE: u64 = 0xD8CA26CBD9BE << 16;
+/// The well-known abort codes used by the compiler. Those confirm
+/// to the error category standard as defined in `errors.move`
```
Comment (brmataptos) [inline third_party/move/move-model/src/well_known.rs:49]: Can you please point to that file?  (Is that third_party/move/move-stdlib/nursery/sources/errors.move?)  If you can point to the doc explaining this, that might also be useful.  This comment isn't quite self-explanatory to me.

Then again, I may just be tired.  Others can approve if they like.

File: third_party/move/move-model/src/well_known.rs [hunk 2]
```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
 /// The well-known name of the first parameter of a method.
 pub const RECEIVER_PARAM_NAME: &str = "self";
 
-/// The well-known abort codes used by the compiler. These are marked
-/// by lowest 6 bytes of a sha256 of the string "Move 2 Abort Code",
-/// appended with two bytes for the error type.
-/// TODO: add a check at runtime that user is not clashing with reserved
-/// codes?
-pub const WELL_KNOWN_ABORT_CODE_BASE: u64 = 0xD8CA26CBD9BE << 16;
+/// The well-known abort codes used by the compiler. Those confirm
```
Comment (vineethk) [inline third_party/move/move-model/src/well_known.rs:48]:
```suggestion
/// The well-known abort codes used by the compiler. These conform
```

File: third_party/move/move-model/src/well_known.rs [hunk 3]
```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
...
+/// to the error category standard as defined in `errors.move`
+/// in the standard library. The lowest three bytes represent
+/// the error category (one byte) and the reason (two bytes).
+/// All compiler generated abort cords use category
```
Comment (vineethk) [inline third_party/move/move-model/src/well_known.rs:52]:
```suggestion
/// All compiler generated abort codes use category
```

File: third_party/move/move-model/src/well_known.rs [hunk 4]
```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
...
+/// `std::error::INTERNAL` (`0xB`). The upper five bytes
+/// are populated with the lowest bytes of the sha256
+/// of the string "Move 2 Abort Code".
+const fn make_abort_code(n: u64) -> u64 {
```
Comment (vineethk) [inline third_party/move/move-model/src/well_known.rs:56]:
```suggestion
const fn make_abort_code(reason: u16) -> u64 {
```

From what I can tell from the comment above, `n` is reason, and is two bytes, so we can use a stricter type.

File: third_party/move/move-model/src/well_known.rs [hunk 5]
```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
...
+    let magic = 0xCA26CBD9BE; // sha256("Move 2 Abort code")
+    (magic << 24) | 0xB << 16 | n
+}
 
 // Used when user omits an abort code.
```
Comment (vineethk) [inline third_party/move/move-model/src/well_known.rs:61]:
```suggestion
// Used when user omits an abort code in an `assert!`.
```

File: third_party/move/move-model/src/well_known.rs [hunk 6]
```diff
@@ -45,15 +45,21 @@ pub const TYPE_NAME_GET_SPEC: &str = "type_name::$get";
...
-pub const UNSPECIFIED_ABORT_CODE: u64 = WELL_KNOWN_ABORT_CODE_BASE;
+pub const UNSPECIFIED_ABORT_CODE: u64 = make_abort_code(0);
 
 // Used when a runtime value falls through a match.
-pub const INCOMPLETE_MATCH_ABORT_CODE: u64 = WELL_KNOWN_ABORT_CODE_BASE | 0x1;
+pub const INCOMPLETE_MATCH_ABORT_CODE: u64 = make_abort_code(1);
```
Comment (vineethk) [inline third_party/move/move-model/src/well_known.rs:65]: I remember seeing the abort codes mentioned in the Move book. We should update them to whatever these new ones are.

**Review by vineethk — APPROVED** [review]
Have a few minor comments, approving assuming they are addressed.

File: third_party/move/move-model/src/well_known.rs (hunk 1 above)
Comment (wrwg) [reply third_party/move/move-model/src/well_known.rs:49]: Added, its `move-stdlib/sources/error.move`. There was a spelling error

File: third_party/move/move-model/src/well_known.rs (hunk 6 above)
Comment (wrwg) [reply third_party/move/move-model/src/well_known.rs:65]: Did already.

File: third_party/move/move-model/src/well_known.rs (hunk 1 above)
Comment (brmataptos) [reply third_party/move/move-model/src/well_known.rs]: It's not useful to resolve a conversation when you've just answered a question.  It means I have to dig around to find out if you answered me.

File: testsuite/forge-cli/src/main.rs [hunk 7]
```diff
@@ -1863,7 +1863,7 @@ fn realistic_env_max_load_test(
             (duration.as_secs() / 10).max(60),
         )
         .add_latency_threshold(3.4, LatencyType::P50)
-        .add_latency_threshold(4.5, LatencyType::P90)
+        .add_latency_threshold(5.2, LatencyType::P90)
```
[the remark this answers is not in the document]
Comment (wrwg) [reply testsuite/forge-cli/src/main.rs:1866]: Look into cicd-testing channel

[the remark this answers is not in the document]
Comment (wrwg) [reply testsuite/forge-cli/src/main.rs:1866]: Or the commit message

**Pull request merged**